internal/gatewayapi/listener.go

```diff
@@ -11,9 +11,18 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/gateway-api/apis/v1beta1"
 
+	configv1a1 "github.com/envoyproxy/gateway/api/config/v1alpha1"
 	"github.com/envoyproxy/gateway/internal/ir"
 )
 
+const (
```

Comment:
imo this code is specific to xds, and should live in the XDS translator, we probably should follow the same style
used in Infra IR where the entire `EnvoyProxy` is passed into the IR and then parsed in the infra layer
https://github.com/envoyproxy/gateway/blob/22bd06b33f914e406bba9ce9690e92cc3f5c3226/internal/ir/infra.go#L36

Reply:
make sense